File: IsingModel_rawpython/IsingEnergy.py
# Import libraries:
import numpy as np

def IsingEnergy(grid, J):
    """
    Compute the energy density of a spin configuration.
    """
    # Calculate the sum of nearest neighbors for each site
    neighbors = np.roll(grid, 1, axis=1) + np.roll(grid, -1, axis=1) + \
                np.roll(grid, 1, axis=0) + np.roll(grid, -1, axis=0)
    
    # Calculate total energy
    energy = -J * np.sum(grid * neighbors) / grid.size
    
    return energy

File: matlab/IsingEnergy.m
function energy = IsingEnergy(grid,J)
% Compute the energy density of a spin configuration.

neighbors = circshift(grid,[0 1]) + circshift(grid,[0 -1]) + ...
            circshift(grid,[1 0]) + circshift(grid,[-1 0]);
energy = -J*sum(sum(grid.*neighbors))/numel(grid);

end
